### DISCUSSION TITLE
Does SKGLView work in Blazor WASM?

### DISCUSSION THREAD

**Discussion by peder1896** [mono/SkiaSharp#2216] [Q&A]
Hi
Does SKGLView work in Blazor WASM? (SKCanvasView works fine)
And if so how to implement?
Thanks

**Comment by mattleibow** [comment]
Yes it does, see the sample:

```razor
<SKGLView OnPaintSurface="OnPaintSurface" IgnorePixelScaling="true" EnableRenderLoop="true" />
```

https://github.com/mattleibow/skiasharp-samples/blob/main/Basic/Blazor/WebAssembly/SkiaSharpSample/Pages/GPU.razor

**Comment by GeorgeS2019** [comment]
@mattleibow  

Do you plan to provide SKGLView sample which supports mouse event e.g. Mouse in, Mouse hover, Mouse Click ?

**Reply by mattleibow** [reply]
No need, the views work exactly the same. All you have to do is change one element type for another. Have you managed to pull down the sample or install the nuget into your project?

**Comment by peder1896** [comment]
Hi Matthew
Thank you for the reply well your code works like a charm,
I had to reecreate a new Project to get the webgl container to work in my Project with SKGLView, could not get it to function in the old project.
Well now i try to apply my logic to the Project and it is so slow compared to the same code as a maui project (and in Maui i use SKCanvasView)
I can send you the code if that helps.
Really strange

Regards Peder

Från: Matthew Leibowitz
Skickat: den 16 augusti 2022 11:43
Till: mono/SkiaSharp
Kopia: peder1896; Author
Ämne: Re: [mono/SkiaSharp] Does SKGLView work in Blazor WASM? (Discussion#2216)

No need, the views work exactly the same. All you have to do is change one element type for another. Have you managed to pull down the sample or install the nuget into your project?

**Comment by peder1896** [comment]
Hi Matthew

When I try to publish my project as a wasm windowsservice with your sample project with skiasharp & skiasharp.views.blazor 2.88.1 I get this error in my browser console

 Failed to find a valid digest in the 'integrity' attribute for resource 'http://localhost:5580/_framework/SkiaSharp.Views.Blazor.dll' with computed SHA-256 integrity 'mnnfdAEuQYbTe9JKRu1rVRILNd37rnaKpjZJE8056T4='. The resource has been blocked.

 

It works fine in debugmode!

I guess I doing something wrong but cant figure out what

Regards Peder

 

 

Från: Matthew Leibowitz ***@***.***>
Svara till: mono/SkiaSharp ***@***.***>
Datum: tisdag 16 augusti 2022 11:43
Till: mono/SkiaSharp ***@***.***>
Kopia: Peder Strauss ***@***.***>, Author ***@***.***>
Ämne: Re: [mono/SkiaSharp] Does SKGLView work in Blazor WASM? (Discussion #2216)

 

No need, the views work exactly the same. All you have to do is change one element type for another. Have you managed to pull down the sample or install the nuget into your project?
